--- openwsman-2.6.2/src/plugins/swig/python/CMakeLists.txt~	2015-10-19 15:27:46.000000000 +0200
+++ openwsman-2.6.2/src/plugins/swig/python/CMakeLists.txt	2015-12-31 16:50:25.747677518 +0100
@@ -9,7 +9,7 @@
 
 FIND_PACKAGE(PythonInterp REQUIRED)
 
-EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import stdout; from distutils import sysconfig; stdout.write(sysconfig.get_python_lib())" OUTPUT_VARIABLE PYTHON_LIB_DIR)
+EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import stdout; from distutils import sysconfig; stdout.write(sysconfig.get_python_lib(plat_specific=True,prefix='/usr'))" OUTPUT_VARIABLE PYTHON_LIB_DIR)
 
 IF (NOT PYTHON_SITE_DIR)
   SET (PYTHON_SITE_DIR ${PYTHON_LIB_DIR})
--- openwsman-2.6.5/bindings/python/CMakeLists.txt.orig	2017-11-28 09:32:15.000000000 +0100
+++ openwsman-2.6.5/bindings/python/CMakeLists.txt	2018-09-03 21:47:58.258854144 +0200
@@ -17,9 +17,9 @@
 ELSE( BUILD_PYTHON3 )
 set(Python_ADDITIONAL_VERSIONS 2.7 2.6)
 ENDIF( BUILD_PYTHON3 )
+FIND_PACKAGE(PythonInterp REQUIRED)
 FIND_PACKAGE(PythonLibs)
 IF (PYTHON_LIBRARY)
-  FIND_PACKAGE(PythonInterp REQUIRED)
   #MESSAGE(STATUS "Found PythonLibs...")
   FIND_PACKAGE(PythonLinkLibs)
   #IF (PYTHON_LINK_LIBS)
@@ -29,7 +29,7 @@
   MESSAGE(STATUS "No Python found")
 ENDIF (PYTHON_LIBRARY)
 
-EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(1))" OUTPUT_VARIABLE PYTHON_LIB_DIR)
+EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(plat_specific=True,prefix='/usr'))" OUTPUT_VARIABLE PYTHON_LIB_DIR)
 STRING(REPLACE "\n" "" PYTHON_LIB_DIR "${PYTHON_LIB_DIR}")
 
 MESSAGE(STATUS "Python executable: ${PYTHON_EXECUTABLE}")
